Sources/Commands/SwiftPackageTool.swift

```diff
@@ -201,12 +201,22 @@ extension SwiftPackageTool {
         var packageName: String?
 
         func run(_ swiftTool: SwiftTool) throws {
-            // FIXME: Error handling.
-            let cwd = localFileSystem.currentWorkingDirectory!
-
-            let packageName = self.packageName ?? cwd.basename
+            guard let cwd = localFileSystem.currentWorkingDirectory else {
+                throw StringError("No Current Working Directory")
+            }
+            var packageName: String
+            var destinationPath = cwd
+            
+            if let optionalPackageName = self.packageName {
+                try localFileSystem.createDirectory(cwd.appending(RelativePath(optionalPackageName)))
+                packageName = optionalPackageName
+                destinationPath = cwd.appending(RelativePath(optionalPackageName))
+            } else {
+                packageName = cwd.basename
+            }
+            
```

Comment:
I wonder if we should make this even more explicit, since IMO in most cases I think users *do not* want the package to be created in the current directory so we should make that "harder" than it is now (the easiest!)

currently proposed:
1. `package init` => crates the package in the current directory
2. `package init --name <name>` => crates a directory `<name>` (if not already exists) and the package within it

alternative to consider/debate:
1. `package init` => errors with under-specified command
2. `package init --name <name>` => crates a directory `<name>` (if not already exists) and the package within it
1. `package init --here` => crates the package in the current directory

Reply:
cc @neonichu @abertelrud

Reply:
I don't see a good reason to make a common command harder to use after it has been in place for many years. Also there is `swift package init --type manifest` which is definitely intended to be used in an existing directory for the use case of adding a manifest to an existing project.

Reply:
Another idea to consider would be to have `--name` be implicit, so it would just be `swift package init MyPackage`

Reply:
Just bikeshedding: I think `--subdirectory <name>` pairs better with `--here` than `--name <name>` does.

Reply:
another approach is to make the simple form (no flags) of `package init` prompt the user for more information and confirmation on the location, like `npm` legacy mode does: https://docs.npmjs.com/cli/v7/commands/npm-init

Reply:
That seems like a good approach.  Only concern (which applies to any change here) is that it would break existing scripts, and in this case there is no tools version we can gate the new behavior on.

Reply:
Prompting seems okay-ish to me. I am not sure why we *have* to change the behavior, but at least in interactive use, we'll not be breaking people. Still feels to me as if `swift package init --type manifest` should be special since the intent is clear.